pkg/logs/internal/launchers/container/tailerfactory/file.go

```diff
@@ -0,0 +1,260 @@
+		Sources: tf.sources,
+	}, nil
+}
+
+// makeDockerFileSource makes a LogSource with Config.Type="file" for a docker container.
+func (tf *factory) makeDockerFileSource(source *sources.LogSource) (*sources.LogSource, error) {
+	containerID := source.Config.Identifier
+
+	path := tf.findDockerLogPath(containerID)
+
+	// check access to the file; if it is not readable, then returning an error will
+	// try to fall back to reading from a socket.
+	f, err := filesystem.OpenShared(path)
+	if err != nil {
+		// (this error already has the form 'open <path>: ..' so needs no further embellishment)
+		return nil, err
+	}
+	f.Close()
+
+	sourceName, serviceName := tf.defaultSourceAndService(source, containersorpods.LogContainers)
+
+	// New file source that inherits most of its parent's properties
+	fileSource := sources.NewLogSource(source.Name, &config.LogsConfig{
+		Type:            config.FileType,
+		Identifier:      containerID,
+		Path:            path,
+		Service:         serviceName,
+		Source:          sourceName,
+		Tags:            source.Config.Tags,
+		ProcessingRules: source.Config.ProcessingRules,
+	})
+
+	// inform the file launcher that it should expect docker-formatted content
+	// in this file
+	fileSource.SetSourceType(sources.DockerSourceType)
+
+	return fileSource, nil
+}
+
+// findDockerLogPath returns a path for the given container.
+func (tf *factory) findDockerLogPath(containerID string) string {
+	switch runtime.GOOS {
+	case "windows":
+		return filepath.Join("c:\\programdata\\docker\\containers", containerID, fmt.Sprintf("%s-json.log", containerID))
+	default: // linux, darwin
+		// this config flag provides temporary support for podman while it is
+		// still recognized by AD as a "docker" runtime.
+		if coreConfig.Datadog.GetBool("logs_config.use_podman_logs") {
+			return filepath.Join(
+				podmanLogsBasePath, "storage/overlay-containers", containerID,
+				"userdata/ctr.log")
+		}
+		return filepath.Join(
+			dockerLogsBasePath, "containers", containerID,
+			fmt.Sprintf("%s-json.log", containerID))
+	}
+}
+
+// makeK8sFileSource makes a LogSource with Config.Type="file" for a container in a K8s pod.
+func (tf *factory) makeK8sFileSource(source *sources.LogSource) (*sources.LogSource, error) {
+	containerID := source.Config.Identifier
+
+	pod, err := tf.workloadmetaStore.GetKubernetesPodForContainer(containerID)
+	if err != nil {
+		return nil, fmt.Errorf("cannot find pod for container %q: %w", containerID, err)
+	}
+
+	var container *workloadmeta.OrchestratorContainer
+	for _, pc := range pod.Containers {
+		if pc.ID == containerID {
+			container = &pc
+			break
+		}
+	}
+
+	if container == nil {
+		// this failure is impossible, as GetKubernetesPodForContainer found
+		// the pod by searching for this container
+		return nil, fmt.Errorf("cannot find container %q in pod %q", containerID, pod.Name)
+	}
+
+	// get the path for the discovered pod and container
+	path := findK8sLogPath(pod, container.Name)
```

Comment:
UT's are filing on windows because all the paths are unix style. Interestingly, we seem to build the `kubelet` tag on windows even though I don't *think* we actually support k8s log collection on windows. 

I think we may need a new build tag to seperate out k8s logging behavior from the `kubelet` since it seems the support matricx for windows does not include logging. 
It seems like in the past we were just getting lucky that the tests worked on windows and the code was never actually called on a real agent - because I can't imagine it would work (unless I'm missing something)

Reply:
Actually - I confirmed that logs on windows is fully supported (or it at least works today)

Reply:
Ive pushed a fix for the windows UTs. 
If CI passes will merge